### PR TITLE
Split BorderService into seperate filters

### LINKED ISSUE #130: Split BorderService into multiple filters
Also, allow a POST to loginConfirm w/o session-id to continue with authentication without an extra 302 roundtrip. It helps in case of performance testing.

### PR DESCRIPTION
**Pull request by maheshkelkar**
- Split into IdentityService, AccessIssuer and UnprotectedService filters
- Allow POST to loginConfirm w/o sessionId to continue with authentication

Fixes #130

### CODE REVIEW

File: core/src/main/scala/com/lookout/borderpatrol/auth/BorderAuth.scala [hunk 1]
```diff
@@ -241,11 +301,10 @@ case class IdentityFilter[A : SessionDataEncoder](store: SessionStore)(implicit
     (for {
       sessionMaybe <- store.get[A](sessionId)
     } yield sessionMaybe.fold[Identity[A]](EmptyIdentity)(s => Id(s.data))) handle {
-      case e => {
+      case e =>
```
Comment (PeterFCaswell) [inline core/src/main/scala/com/lookout/borderpatrol/auth/BorderAuth.scala]: Why did you remove the curly braces? Doesn't that change the code? Does EmptyIdentity only apply to the case e or to every case?

Comment (maheshkelkar) [inline core/src/main/scala/com/lookout/borderpatrol/auth/BorderAuth.scala]: These braces are not needed.

Comment (PeterFCaswell) [inline core/src/main/scala/com/lookout/borderpatrol/auth/BorderAuth.scala]: Not needed by the compiler, but maybe needed for human comprehension? I always find the scala terse-is-good and no-extra-characters paradigms to make it difficult to read. But whatever.
